gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AgentDirectoryMapper.java

```diff
@@ -71,9 +71,10 @@ private void mapAgent(RCCTMT120101UK01Agent agent, List<DomainResource> agentRes
         var agentOrganization = agent.getAgentOrganization();
         var representedOrganization = agent.getRepresentedOrganization();
         var resourceId = agent.getId().get(0).getRoot();
+        var gpNumber = agent.getId().size() > 1 ? agent.getId().get(1).getExtension() : "";
```

Comment:
I think using !isEmpty() here is more readable, and i believe is best practice as in some collections .size will have to count all the items before checking, whereas isEmpty() will short-circuit on a single value found

Reply:
Also gpNumber can only come from one source, so the string value should really be null not an empty string as there is no intention of adding to it at a later point in the code

Reply:
while I agree about the isEmpty, I intentionally don't assign gpNumber null value as it is not a good practice I think and is prone to NPE errors in different places. "" string is more safer and doesn't cause any side effects.

Reply:
actually we need to use agent.getId().size() > 1 as in this case I need to specifically check that collection has more than one element stored...

Reply:
Oh sorry yest I misread as >= 1 :), thats fine!